docs/design/aws_under_the_hood.md

```diff
@@ -0,0 +1,271 @@
+* Associates the subnet to the route table
+* Creates security groups for the master node (`kubernetes-master-<clusterid>`)
+  and the minion nodes (`kubernetes-minion-<clusterid>`)
+* Configures security groups so that masters & minions can intercommunicate,
+  and opens SSH to the world on master & minions, and opens port 443 to the
+world on the master (for the HTTPS API endpoint)
+* Creates an EBS volume for the master node of size `MASTER_DISK_SIZE` and type
+  `MASTER_DISK_TYPE`
+* Launches a master node with a fixed IP address (172.20.0.9), with the
+  security group, IAM credentials etc.  An instance script is used to pass
+vital configuration information to Salt.  The hope is that over time we can
+reduce the amount of configuration information that must be passed in this way.
+* Once the instance is up, it attaches the EBS volume & sets up a manual
+  routing rule for the internal network range (`MASTER_IP_RANGE`, defaults to
+10.246.0.0/24)
+* Creates an auto-scaling launch-configuration and group for the minions.  The
+  name for both is `<KUBE_AWS_INSTANCE_PREFIX>-minion-group`, defaults to
+`kubernetes-minion-group`.  The auto-scaling group has size min & max both set
+to `NUM_MINIONS`.  You can change the size of the auto-scaling group to add or
+remove minions (directly though the AWS API/Console).  The minion nodes
+self-configure: they come up, run Salt with the stored configuration; connect
+to the master and are assigned an internal CIDR; the master configures the
+route-table with the minion CIDR.  The script does health-check the minions,
+but this is a self-check, it is not required.
+
+If attempting this configuration manually, I highly recommend following along
+with the kube-up script, and being sure to tag everything with a
+`KubernetesCluster`=`<clusterid>` tag.  Also, passing the right configuration
+options to Salt when not using the script is tricky: the plan here is to
+simplify this by having Kubernetes take on more node configuration, and even
+potentially remove Salt altogether.
+
+
+## Manual infrastructure creation
+
+While this work is not yet complete, advanced users may choose to create (some)
+AWS objects themselves, and still make use of the kube-up script (to configure
+Salt, for example).
+
+* `AWS_S3_BUCKET` will use an existing S3 bucket
+* `VPC_ID` will reuse an existing VPC
+* `SUBNET_ID` will reuse an existing subnet
+* If your route table is tagged with the correct `KubernetesCluster`, it will
+  be reused
+* If your security groups are appropriately named, they will be reused.
+
+Currently there is no way to do the following with kube-up.  If these affect
+you, please open an issue with a description of what you're trying to do (your
+use-case) and we'll see what we can do:
+
+* Use an existing AWS SSH key with an arbitrary name
+* Override the IAM credentials in a sensible way (but this is in-progress)
+* Use different security group permissions
+* Configure your own auto-scaling groups
+
```

Comment:
Insert here (move from above):

If any of the above items apply to your situation, open an issue to request an enhancement to the kube-up script. You should provide a complete description of the use-case, including all the details around what you want to accomplish.